rastervision/data/scene_config.py

```diff
@@ -107,39 +107,31 @@ def create_local(self, tmp_dir):
     def to_builder(self):
         return SceneConfigBuilder(self)
 
-    def update_for_command(self, command_type, experiment_config,
-                           context=None):
+    def update_for_command(self,
+                           command_type,
+                           experiment_config,
+                           context=None,
+                           io_def=None):
         if context is None:
             context = []
         context = context + [self]
-        io_def = rv.core.CommandIODefinition()
+        io_def = io_def or rv.core.CommandIODefinition()
 
-        b = self.to_builder()
-
-        (new_raster_source,
-         sub_io_def) = self.raster_source.update_for_command(
-             command_type, experiment_config, context)
-        io_def.merge(sub_io_def)
-        b = b.with_raster_source(new_raster_source)
+        self.raster_source.update_for_command(command_type, experiment_config,
+                                              context, io_def)
 
         if self.label_source:
-            (new_label_source,
-             sub_io_def) = self.label_source.update_for_command(
-                 command_type, experiment_config, context)
-            io_def.merge(sub_io_def)
-            b = b.with_label_source(new_label_source)
+            self.label_source.update_for_command(
```

Comment:
Don't we need to capture the `io_def` returned by this call and then merge with `io_def`?

Reply:
No, because now with the mutable io_def, the are just passed forward and configs all sequentially modify the same io_def instance. So when a label_source adds to the io_def, that's the same io_def that is further modified by the label_store etc

Reply:
Oh, right. So I guess that means we don't need to return io_def then.

Reply:
Actually that's not quite true because io_def defaults to None in the parameter list.